### PR TITLE
Add XAI detection hook

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
### Summary

- Added `DetClassProbabilityMapHook` for detection models: ATSS, SSD, YOLOX
- Added tests for hook
- Small updates in `setup_guide.md`

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by GalyaZalesskaya** [comment]
After merging `v2` code-quility checks are failing

File: src/otx/core/model/entity/detection.py [hunk 1]
```diff
@@ -59,7 +59,69 @@ def forward_tiles(self, inputs: TileBatchDetDataEntity) -> DetBatchPredEntity:
...
+        """Gets the anchor configuration from model."""
+        anchor_generator = getattr(self.model.bbox_head, "prior_generator", None)
+        if anchor_generator is not None:
+            if hasattr(anchor_generator, "num_base_anchors"):
+                num_anchors = anchor_generator.num_base_anchors
+            else:
+                num_anchors = anchor_generator.num_base_priors
+        else:
+            num_anchors = [1] * 10
+        return num_anchors
```
Comment (vinnamkim) [inline src/otx/core/model/entity/detection.py]:
```suggestion
        if anchor_generator := getattr(self.model.bbox_head, "prior_generator", None):
            return (
                 anchor_generator.num_base_anchors 
                 if hasattr(anchor_generator, "num_base_anchors")
                 else anchor_generator.num_base_priors
            )

        return [1] * 10
```
Please be careful not to create many nested lines as possible

Comment (GalyaZalesskaya) [reply src/otx/core/model/entity/detection.py]: Great solution, thank you

**Review by vinnamkim — DISMISSED** [review]

**Comment by negvet** [comment]
I believe with this PR we can extend the scope for https://github.com/openvinotoolkit/training_extensions/blob/96b6b2a5c9c572071bad428f664879a974063aa1/tests/integration/cli/test_cli.py#L149-L167

What do you think?

**Comment by negvet** [comment]
Also please manually check if it works for some random model, if not yet

File: for_developers/setup_guide.md [hunk 2]
```diff
@@ -19,11 +26,17 @@ pip install -e .
...
+```
+pip install --upgrade pip
+```
+
 ### With PIP & 'otx install'
 
 ```console
 # Create venv
-python -m venv venv
+python3.10 -m venv venv
```
Comment (sovrasov) [inline for_developers/setup_guide.md]: It's 3.9 already in the requirements above, so I think we shouldn't specify the exact version here

Comment (GalyaZalesskaya) [reply for_developers/setup_guide.md]: Done, thank you

**Comment by GalyaZalesskaya** [comment]
> I believe with this PR we can extend the scope for integration/test_cli.py

Thank you for the comment, I extended scope of e2e tests, here is the list of the tests
image

**Review by negvet — APPROVED** [review]
LGTM

**Review by eunwoosh — APPROVED** [review]

**Review by MarkByun — APPROVED** [review]
LGTM

**Review by sungmanc — APPROVED** [review]

**Pull request merged**